src/caffe/common.cpp

```diff
@@ -111,22 +111,29 @@ Caffe::Caffe()
 #endif
     random_generator_(),
     mode_(Caffe::CPU), solver_count_(1), root_solver_(true) {
+  // Try to create a cuda stream.
+  if (cudaStreamCreate(&thread_stream_) != cudaSuccess) {
+    LOG(ERROR) << "Cannot create cuda stream. Cuda stream won't be available.";
+  }
```

Comment:
Same here: I am not sure we want to do that by dafault, why is default stream not good ?